File: Microsoft Office Developer Documentation Team/Office-2010-Deploying-Multiple-Office-2010-3887/[VB]-Office 2010 Deploying Multiple Office 2010 Projects in One Package/VB/ExcelSmartTagAddIn/Ribbon1.vb
Imports Microsoft.Office.Tools.Ribbon

Public Class Ribbon1

  Private Sub Ribbon1_Load(ByVal sender As System.Object, ByVal e As RibbonUIEventArgs) Handles MyBase.Load

  End Sub

  Private Sub TemperatureEditBox_TextChanged(ByVal sender As System.Object, ByVal e As Microsoft.Office.Tools.Ribbon.RibbonControlEventArgs) Handles TemperatureEditBox.TextChanged
    InsertButton.Enabled = (TemperatureEditBox.Text.Length > 0)
  End Sub

  Private Sub InsertButton_Click(ByVal sender As System.Object, ByVal e As Microsoft.Office.Tools.Ribbon.RibbonControlEventArgs) Handles InsertButton.Click
    Dim temp As String = TemperatureEditBox.Text
    Select Case ScaleDropDown.SelectedItem.Label
      Case "Fahrenheit"
        temp &= "°F"
      Case "Centigrade"
        temp &= "°C"
    End Select
    Dim _cell = Globals.ThisAddIn.Application.ActiveCell
    _cell.Value = temp
  End Sub

End Class

File: Microsoft Office Developer Documentation Team/Office-2010-Deploying-Multiple-Office-2010-3887/[C#]-Office 2010 Deploying Multiple Office 2010 Projects in One Package/C#/ExcelSmartTagAddIn/Ribbon1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Office.Tools.Ribbon;

namespace ExcelSmartTagAddIn
{
  public partial class Ribbon1
  {
    private void Ribbon1_Load(object sender, RibbonUIEventArgs e)
    {

    }

    private void TemperatureEditBox_TextChanged(object sender, RibbonControlEventArgs e)
    {
      InsertButton.Enabled = (TemperatureEditBox.Text.Length > 0);
    }

    private void InsertButton_Click(object sender, RibbonControlEventArgs e)
    {
      String temp = TemperatureEditBox.Text;
      switch (ScaleDropDown.SelectedItem.Label)
      {
        case "Fahrenheit":
          temp += "°F";
          break;

        case "Centigrade":
          temp += "°C";
          break;
      }
      var cell = Globals.ThisAddIn.Application.ActiveCell;
      cell.Value = temp;
    }

  }
}
